src/graph/src/graph.ts

```diff
@@ -393,6 +394,18 @@ export class Graph implements GraphLike {
     toNode.maybeSetConfusedManifest(spec, manifest)
 
     this.addEdge(depType, spec, fromNode, toNode)
+
+    // populate resolution cache if applicable
+    const f = getResolutionCacheKey(
+      spec.final,
+      fromNode.location,
+      extra || '',
+    )
+    this.resolutions.set(f, toNode)
+    const rrev = this.resolutionsReverse.get(toNode) ?? new Set()
+    rrev.add(f)
+    this.resolutionsReverse.set(toNode, rrev)
+
```

Comment:
moved from `addNode` to `placePackage` which simplified a bit the logic but most importantly allows for using the proper `getResolutionCacheKey` method instead of relying on the stringified spec as a key.